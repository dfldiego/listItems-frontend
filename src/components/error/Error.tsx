type Props = {
  error: string;
  loading: boolean;
};

const Error = ({ error, loading }: Props) => {
  return (
    <>
      {!loading && error.length > 0 && (
        <p className="error mt5 errorGrid">{error}</p>
      )}
    </>
  );
};

export default Error;
