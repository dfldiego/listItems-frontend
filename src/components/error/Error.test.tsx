import { render, screen } from "@testing-library/react";
import Error from "./Error";

const PROPS_LOADING_FALSE = {
  error: "errorTest",
  loading: false,
};

const PROPS_LOADING_TRUE = {
  error: "errorTest",
  loading: true,
};

describe("<Error>", () => {
  test("should render correctly", () => {
    const { container } = render(<Error {...PROPS_LOADING_FALSE} />);
    expect(container).toMatchSnapshot();
  });

  test("should render the error message", () => {
    render(<Error {...PROPS_LOADING_FALSE} />);
    expect(screen.getByText(/errorTest/i)).toBeInTheDocument();
  });

  test("should not render the error message", () => {
    render(<Error {...PROPS_LOADING_TRUE} />);
    expect(screen.queryByText(/errorTest/i)).not.toBeInTheDocument();
  });
});
