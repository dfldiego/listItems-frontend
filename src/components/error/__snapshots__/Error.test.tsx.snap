// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error> should render correctly 1`] = `
<div>
  <p
    class="error mt5 errorGrid"
  >
    errorTest
  </p>
</div>
`;
